lnrpc/routerrpc/router_server.go

```diff
@@ -888,8 +893,7 @@ func (s *Server) HtlcInterceptor(stream Router_HtlcInterceptorServer) error {
 	}
 	defer atomic.CompareAndSwapInt32(&s.forwardInterceptorActive, 1, 0)
 
-	// run the forward interceptor.
-	return newForwardInterceptor(s, stream).run()
+	return s.forwardInterceptor.attachStream(stream)
```

Comment:
One thing that I had to think of is how to deal with the interceptor crashing and missing a packet? On reconnect, lnd won't resend but still wait for the resolution. Does the interceptor need to ack packets to make it safe?

Reply:
Oh, yeah, good point. Perhaps HTLCs can be marked as "needs notification" until a `HOLD` action is received from the interceptor.

Reply:
Yes that would be good. Always expect a response. And on reconnect, replay everything in the wait for ack queue?

Reply:
Or simplify this and just always replay everything that's pending when the interceptor reconnects. This can simplify things on the interceptor side too, because it doesn't need to persist state for the acked htlcs.

Reply:
Yeah, interceptors need to be able to handle replays anyway and it would make everything quite simpler.